### DISCUSSION TITLE
Use cases for swizzling theme components?

### DISCUSSION THREAD

**Discussion by yangshun** [facebook/docusaurus#2790] [General]
Following up on https://github.com/facebook/docusaurus/pull/2743:

What are your common use cases for [swizzling of components](https://v2.docusaurus.io/docs/using-themes/#swizzling-theme-components)? We're coming up with new APIs for theme components so that you can customize them without swizzling. Your feedback will help us shape those APIs.

Let us know in the comments!

**Comment by lex111** [comment]
Offtop: I think such questions are best asked on our Discord/Twitter accounts, no one will know about it here.

**Reply by yangshun** [reply]
I think discussions is the correct place to do that. Discussions is a new GitHub feature and I'm not sure if people can see it.

**Reply by omry** [reply]
I can now see it. how do you enable this?

**Reply by slorber** [reply]
@omry need to verify but it's a new GH feature and a few months ago we had to ask Github to activate it, so it may not be available to all repos yet

**Comment by lex111** [comment]
I meant it was necessary to invite to this discussion using the
Twitter/Discord account (to let people know about it).

вс, 24 мая 2020 г. в 15:10, Yangshun Tay <notifications@github.com>:

**Comment by slorber** [comment]
Hi,

I'm not fan at all of this swizzling thing (also known as shadowing in gatsby).

I'd rather have an approach that allows to override components using context, in the same way MDXProvider works.

Just expressed some thoughts here, hope to get some feedback: https://twitter.com/sebastienlorber/status/1264599806988767232

**Comment by slorber** [comment]
Chris Biscardi answered me that it's because we don't want to bundle 2 variants of the same code, while only using 1 at runtime.

Just saw dynamic imports in a provider in the Dokz project documentation, that could be a solution to this problem!

```
        <ClassicThemeProvider themeComponents={{Navbar: () => import('./Navbar')}}>
            {children}
        </ClassicThemeProvider>
```
